portal/models/observation.py

```diff
@@ -0,0 +1,112 @@
+from datetime import datetime
+
+from ..database import db
+from ..date_tools import FHIR_datetime, as_fhir
+from .performer import Performer
+from .value_quantity import ValueQuantity
+
+
+class Observation(db.Model):
+    __tablename__ = 'observations'
+    id = db.Column(db.Integer, primary_key=True)
+    issued = db.Column(db.DateTime, default=datetime.utcnow)
+    status = db.Column(db.String(80))
+    codeable_concept_id = db.Column(
+        db.ForeignKey('codeable_concepts.id'), nullable=False)
+    value_quantity_id = db.Column(
+        db.ForeignKey('value_quantities.id'), nullable=False)
+    codeable_concept = db.relationship(
+        'CodeableConcept', cascade="save-update")
+    value_quantity = db.relationship('ValueQuantity')
+    performers = db.relationship(
+        'Performer', lazy='dynamic', cascade="save-update",
+        secondary="observation_performers", backref=db.backref('observations'))
+
+    def __str__(self):
+        """Print friendly format for logging, etc."""
+        at = ' at {0.issued}'.format(self) if self.issued else ''
+        status = ' with status {0.status}'.format(self) if self.status else ''
+        return (
+            "Observation {0.codeable_concept} {0.value_quantity}{at}"
+            "{status}".format(self, at=at, status=status))
+
+    def as_fhir(self):
+        """Return self in JSON FHIR formatted string"""
+        fhir = {"resourceType": "Observation"}
+        if self.issued:
+            fhir['issued'] = as_fhir(self.issued)
+        if self.status:
+            fhir['status'] = self.status
+        fhir['id'] = self.id
```

Comment:
Slightly easier to read if these were folded into the initial `fhir` dict initialization, before any of the conditional code